kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveRestoreConsumer.java

```diff
@@ -0,0 +1,74 @@
+package dev.responsive.kafka.clients;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.OptionalLong;
+import java.util.function.Function;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+public class ResponsiveRestoreConsumer<K, V> extends DelegatingConsumer<K, V> {
```

Comment:
nit: javadoc explaining the nuances of the interaction between this and the streams restore protocol would be helpful